<html>
    <head>
    <title>Error - try, catch, finally</title>
</head>
<body>
    <script>

        // consol.log("안녕하세요?"); // error
        // console.log("오늘은 즐거운 금요일입니다."); // 출력 X why? 위의 코드에서 error 발생

        // try {
        //     consol.log("안녕하세요?");    
        // } catch (error) { // error가 발생하지 않는다면 이 안의 문장 실행 X
        //     console.log(error.name); // 발생한 error의 name
        //     console.log(error.message); // 발생한 error의 message
        // }
        // console.log("오늘은 즐거운 금요일입니다."); // 출력 O, error 발생하면 -> 예외 처리

        // throw를 사용하여 강제로 Exception을 발생
        var x = "";
        try {
            if(x === "") throw "Empty"; // 문법적인 error는 아니지만 throw해서 강제로 Exception
            var y = x + 3;
            console.log(y);
        } catch (error) {
            console.log(`x is ${error}`);   
        }
        finally { // 반드시 사용해야 할 문장
            console.log("error 유무와 상관없이 무조건 실행됨.");
        }

    </script>
</body>
</html>